### ISSUE TITLE
Using composer Latest on v3.2.0

### ISSUE THREAD

**Issue by clement360** [cloudfoundry/php-buildpack#70]
If I use "COMPOSER_VERSION": "latest" in options.json in version 3.2 of the build pack I get the following error: 

2015-05-06T10:36:46.02-0400 [STG/0]      OUT Downloaded [https://pivotal-buildpacks.s3.amazonaws.com/php/binaries/lucid/php/5.4.39/php-cli-5.4.39.tar.gz] to [/tmp]
2015-05-06T10:36:46.51-0400 [STG/0]      ERR Traceback (most recent call last):
2015-05-06T10:36:46.51-0400 [STG/0]      ERR   File "/tmp/buildpacks/php-buildpack/scripts/compile.py", line 51, in <module>
2015-05-06T10:36:46.51-0400 [STG/0]      ERR     .from_build_pack('lib/additional_commands')
2015-05-06T10:36:46.51-0400 [STG/0]      ERR   File "/tmp/buildpacks/php-buildpack/lib/build_pack_utils/builder.py", line 189, in extensions
2015-05-06T10:36:46.51-0400 [STG/0]      ERR     process_extension(path, ctx, 'compile', process, args=[self])
2015-05-06T10:36:46.51-0400 [STG/0]      ERR   File "/tmp/buildpacks/php-buildpack/lib/build_pack_utils/utils.py", line 69, in process_extension
2015-05-06T10:36:46.51-0400 [STG/0]      ERR     success(getattr(extn, to_call)(*args))
2015-05-06T10:36:46.51-0400 [STG/0]      ERR   File "/tmp/buildpacks/php-buildpack/extensions/composer/extension.py", line 400, in compile
2015-05-06T10:36:46.51-0400 [STG/0]      ERR     return composer.compile(install)
2015-05-06T10:36:46.51-0400 [STG/0]      ERR   File "/tmp/buildpacks/php-buildpack/lib/extension_helpers.py", line 154, in compile
2015-05-06T10:36:46.51-0400 [STG/0]      ERR     self._compile(install)
2015-05-06T10:36:46.51-0400 [STG/0]      ERR   File "/tmp/buildpacks/php-buildpack/extensions/composer/extension.py", line 163, in _compile
2015-05-06T10:36:46.51-0400 [STG/0]      ERR     self.install()
2015-05-06T10:36:46.51-0400 [STG/0]      ERR   File "/tmp/buildpacks/php-buildpack/extensions/composer/extension.py", line 187, in install
2015-05-06T10:36:46.51-0400 [STG/0]      ERR     extract=False)
2015-05-06T10:36:46.51-0400 [STG/0]      ERR   File "/tmp/buildpacks/php-buildpack/lib/build_pack_utils/cloudfoundry.py", line 176, in _install_binary_from_manifest
2015-05-06T10:36:46.51-0400 [STG/0]      ERR     self._dwn.download(url, self._ctx['TMPDIR'])
2015-05-06T10:36:46.51-0400 [STG/0]      ERR   File "/tmp/buildpacks/php-buildpack/lib/build_pack_utils/downloads.py", line 49, in download
2015-05-06T10:36:46.51-0400 [STG/0]      ERR     raise RuntimeError("Could not download dependency: %s" % url)
2015-05-06T10:36:46.51-0400 [STG/0]      <b>ERR RuntimeError: Could not download dependency: https://getcomposer.org/composer.phar</b>
2015-05-06T10:36:46.54-0400 [STG/0]      OUT Staging failed: Buildpack compilation step failed
2015-05-06T10:36:47.48-0400 [API/9]      ERR Encountered error: App staging failed in the buildpack compile phase

If you remove the composer : latest option everything works great. I feel this error may confuse people using the composer_version tag.

**Comment by dmikusa-pivotal** [comment]
This is a known problem.  Version 3.1.0 of the build pack was the last one where setting the COMPOSER_VERSION to "latest" work to pull in a dev release of Composer.  You can revert to that version of the build pack, don't set COMPOSER_VERSION which defaults to the latest published release or you can set a specific published release.  In most cases, you shouldn't need to set this.

**Comment by flavorjones** [comment]
We've got a story in the Tracker project to address this: https://www.pivotaltracker.com/story/show/92519522.

**Comment by styeung** [comment]
@clement360: We've added back the ability to set COMPOSER_VERSION to 'latest' with uncached buildpacks.  This change was made on our develop branch, and you can expect it to be in our next release.  Thanks!

**Comment by flavorjones** [comment]
Verified `develop` addresses this. Great!

**Comment by dmikusa-pivotal** [comment]
I think this is broken again.

```
2015-05-21T08:46:01.72-0400 [STG/0]      ERR 2015-05-21 12:46:01,720 [INFO] zips - Extracting [/tmp/php-cli-5.4.39.tar.gz] into [/tmp/staged/app/php]
2015-05-21T08:46:02.00-0400 [STG/0]      ERR 2015-05-21 12:46:02,004 [DEBUG] utils - line #183 in /var/vcap/data/dea_next/admin_buildpacks/8d56f157-755c-4873-94e3-d2a2d47cc0ac_6d5d8b9f582367562f176efc3aa9396fb64f44f1/extensions/composer/extension.py, "install" is setting [COMPOSER_DOWNLOAD_URL] = [https://getcomposer.org/composer.phar]
2015-05-21T08:46:02.00-0400 [STG/0]      ERR 2015-05-21 12:46:02,005 [DEBUG] cloudfoundry - Using python downloader.
2015-05-21T08:46:02.00-0400 [STG/0]      ERR 2015-05-21 12:46:02,005 [DEBUG] downloads - Loaded proxy handlers [{}]
2015-05-21T08:46:02.00-0400 [STG/0]      ERR 2015-05-21 12:46:02,005 [DEBUG] cloudfoundry - Installing binary from manifest [https://getcomposer.org/composer.phar]
2015-05-21T08:46:02.23-0400 [STG/0]      ERR Traceback (most recent call last):
2015-05-21T08:46:02.23-0400 [STG/0]      ERR   File "/var/vcap/data/dea_next/admin_buildpacks/8d56f157-755c-4873-94e3-d2a2d47cc0ac_6d5d8b9f582367562f176efc3aa9396fb64f44f1/scripts/compile.py", line 51, in <module>
2015-05-21T08:46:02.23-0400 [STG/0]      ERR     .from_build_pack('lib/additional_commands')
2015-05-21T08:46:02.23-0400 [STG/0]      ERR   File "/var/vcap/data/dea_next/admin_buildpacks/8d56f157-755c-4873-94e3-d2a2d47cc0ac_6d5d8b9f582367562f176efc3aa9396fb64f44f1/lib/build_pack_utils/builder.py", line 189, in extensions
2015-05-21T08:46:02.23-0400 [STG/0]      ERR     process_extension(path, ctx, 'compile', process, args=[self])
2015-05-21T08:46:02.23-0400 [STG/0]      ERR   File "/var/vcap/data/dea_next/admin_buildpacks/8d56f157-755c-4873-94e3-d2a2d47cc0ac_6d5d8b9f582367562f176efc3aa9396fb64f44f1/lib/build_pack_utils/utils.py", line 69, in process_extension
2015-05-21T08:46:02.23-0400 [STG/0]      ERR     success(getattr(extn, to_call)(*args))
2015-05-21T08:46:02.23-0400 [STG/0]      ERR   File "/var/vcap/data/dea_next/admin_buildpacks/8d56f157-755c-4873-94e3-d2a2d47cc0ac_6d5d8b9f582367562f176efc3aa9396fb64f44f1/extensions/composer/extension.py", line 400, in compile
2015-05-21T08:46:02.23-0400 [STG/0]      ERR     return composer.compile(install)
2015-05-21T08:46:02.23-0400 [STG/0]      ERR   File "/var/vcap/data/dea_next/admin_buildpacks/8d56f157-755c-4873-94e3-d2a2d47cc0ac_6d5d8b9f582367562f176efc3aa9396fb64f44f1/lib/extension_helpers.py", line 154, in compile
2015-05-21T08:46:02.23-0400 [STG/0]      ERR     self._compile(install)
2015-05-21T08:46:02.23-0400 [STG/0]      ERR   File "/var/vcap/data/dea_next/admin_buildpacks/8d56f157-755c-4873-94e3-d2a2d47cc0ac_6d5d8b9f582367562f176efc3aa9396fb64f44f1/extensions/composer/extension.py", line 163, in _compile
2015-05-21T08:46:02.23-0400 [STG/0]      ERR     self.install()
2015-05-21T08:46:02.23-0400 [STG/0]      ERR   File "/var/vcap/data/dea_next/admin_buildpacks/8d56f157-755c-4873-94e3-d2a2d47cc0ac_6d5d8b9f582367562f176efc3aa9396fb64f44f1/extensions/composer/extension.py", line 187, in install
2015-05-21T08:46:02.23-0400 [STG/0]      ERR     extract=False)
2015-05-21T08:46:02.23-0400 [STG/0]      ERR   File "/var/vcap/data/dea_next/admin_buildpacks/8d56f157-755c-4873-94e3-d2a2d47cc0ac_6d5d8b9f582367562f176efc3aa9396fb64f44f1/lib/build_pack_utils/cloudfoundry.py", line 176, in _install_binary_from_manifest
2015-05-21T08:46:02.23-0400 [STG/0]      ERR     self._dwn.download(url, self._ctx['TMPDIR'])
2015-05-21T08:46:02.23-0400 [STG/0]      ERR   File "/var/vcap/data/dea_next/admin_buildpacks/8d56f157-755c-4873-94e3-d2a2d47cc0ac_6d5d8b9f582367562f176efc3aa9396fb64f44f1/lib/build_pack_utils/downloads.py", line 49, in download
2015-05-21T08:46:02.23-0400 [STG/0]      ERR     raise RuntimeError("Could not download dependency: %s" % url)
2015-05-21T08:46:02.24-0400 [STG/0]      ERR RuntimeError: Could not download dependency: https://getcomposer.org/composer.phar
2015-05-21T08:46:02.27-0400 [STG/0]      OUT Staging failed: Buildpack compilation step failed
2015-05-21T08:46:02.43-0400 [API/6]      ERR encountered error: App staging failed in the buildpack compile phase
```

I get the above output when trying to push to PWS using the default PHP build pack which is currently 3.2.1.  I get the same if I override the build pack URL to the master branch.

The [fix for this](https://github.com/cloudfoundry/php-buildpack/blob/982c992f24ce4f018fb483cc6d65ac3f1d70a9f7/extensions/composer/extension.py#L181-L193) introduced code that would install the latest version using `install_binary_direct()`, but in master it seems to be [back to trying](https://github.com/cloudfoundry/php-buildpack/blob/master/extensions/composer/extension.py#L180-L187) to use `install_binary_from_manifest()`.  Not sure if this is the problem, but thought I'd point it out.

Thanks!

**Comment by dmikusa-pivotal** [comment]
Sorry, I think I may have misunderstood.  I thought this was merged into master and released.  I looked a bit further and it only seems to be in the `develop` branch.  Is that correct?  If so, sorry for reopening this.

**Comment by crawsible** [comment]
Hiya Dan; I don't think we've cut a new release of the PHP buildpack since re-adding `COMPOSER_VERSION: latest` support, so it'll be on the `develop` branch until then. We changed our workflow somewhat recently to only merge changes into master when cutting a new release, but if it's in develop, it'll be in master for the next release. Sorry for the confusion, dude!

**Comment by dmikusa-pivotal** [comment]
Thanks for confirming!

**Comment by flavorjones** [comment]
This fix is in v3.2.2 of the php-buildpack. Sorry for the confusion.
